pkg/resource/deploy/deployment_executor.go

```diff
@@ -278,7 +266,7 @@ func (ex *deploymentExecutor) Execute(callerCtx context.Context, opts Options, p
 	// valid.  We have to do this *after* performing the steps as the target list may have referred
 	// to a resource that was created in one of the steps.
 	if res == nil {
-		res = ex.checkTargets(opts.UpdateTargets, OpUpdate)
+		res = ex.checkTargets(opts.Targets)
```

Comment:
https://github.com/pulumi/pulumi/pull/13010/files#r1212056905

@Frassle we're using the call down here to check targets so the step generator has already been run.

It seems to me that we'd want to check after the `stepGenerator` not after the `stepExecutor`, but that's current behavior...

I think adding the PreRunChecks and PostRunChecks functionality to `pkg/engine` would improve this.